core/trino-main/src/main/java/io/trino/sql/planner/optimizations/AddExchanges.java

```diff
@@ -414,6 +414,18 @@ public PlanWithProperties visitTopN(TopNNode node, PreferredProperties preferred
                     return rebaseAndDeriveProperties(node, child);
                 case PARTIAL:
                     child = planChild(node, PreferredProperties.any());
+                    // If source is pre-sorted, partial topN can be replaced with partial limit N.
```

Comment:
@kasiafi Do we might need to set `withOrderSensitive` in `LocalAddExchanges` for `TopN` with preordered input. 

I don't think it's the case because pre-ordering property is relevant within local stream.